### PR TITLE
SystemCatalog PRIMARYKEYS return wrong column names for matviews

### PR DESCRIPTION
**Pull request by dm89**
fixed reference to primary key for materialized views

### CODE REVIEW

File: src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java [hunk 1]
```diff
@@ -572,7 +572,7 @@ VoltTable getPrimaryKeys()
                         results.addRow(null,
                                        null, // table schema
                                        table.getTypeName(), // table name
-                                       column.getTypeName(), // column name
+                                       c.getTypeName().equals("MATVIEW_PK_CONSTRAINT") ? column.getColumn().getTypeName() : column.getTypeName(), // column name
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: Please pull this logic out of the `addRow()` function call and make it a variable like `columnName`. It is too long and does not have comments to explain why we are doing this.

Please replace `"MATVIEW_PK_CONSTRAINT"` with
`HSQLInterface.AUTO_GEN_MATVIEW + HSQLInterface.AUTO_GEN_MATVIEW_CONST`

Comment (dm89) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: Actually, just the `HSQLInterface.AUTO_GEN_MATVIEW_CONST` would suffice

File: src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java [hunk 2]
```diff
@@ -569,10 +570,20 @@ VoltTable getPrimaryKeys()
                 {
                     for (ColumnRef column : c.getIndex().getColumns())
                     {
+                            String columnName;
+
+                            // ENG-6927
+                            // if the index type == "MATVIEW_PK_CONSTRAINT", the column name for the materialized view is obtained
+                            // after getting the column information and then its type name - one more level of indirection
+                            if( c.getTypeName().equals(HSQLInterface.AUTO_GEN_MATVIEW_CONST) ) {
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: space after `if`

File: src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java [hunk 3]
```diff
@@ -569,10 +570,20 @@ VoltTable getPrimaryKeys()
                 {
                     for (ColumnRef column : c.getIndex().getColumns())
                     {
+                            String columnName;
+
+                            // ENG-6927
+                            // if the index type == "MATVIEW_PK_CONSTRAINT", the column name for the materialized view is obtained
+                            // after getting the column information and then its type name - one more level of indirection
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: Rephrase it

File: src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java [hunk 4]
```diff
@@ -569,10 +570,19 @@ VoltTable getPrimaryKeys()
...
                     {
+                            String columnName;
+
+                            // if the index name is "MATVIEW_PK_CONSTRAINT", the column name for the materialized view is obtained
+                            // from its column information - ENG-6927
+                            if (c.getTypeName().equals(HSQLInterface.AUTO_GEN_MATVIEW_CONST) ) {
+                                    columnName = column.getColumn().getTypeName();
+                            } else {
+                                    columnName = column.getTypeName();
+                            }
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: Fix the indentation.

Comment (yzhang1991) [inline src/frontend/org/voltdb/JdbcDatabaseMetaDataGenerator.java]: Fix the indentation, please.

**Pull request merged**